frontend/src/features/documents/documentUpload/DocumentUploadContainer.tsx

```diff
@@ -0,0 +1,57 @@
+import { DocumentRelationshipType } from 'constants/documentRelationshipType';
+import { useApiDocuments } from 'hooks/pims-api/useApiDocuments';
+import useIsMounted from 'hooks/useIsMounted';
+import { Api_DocumentType, Api_UploadRequest } from 'models/api/Document';
+import { useEffect, useState } from 'react';
+
+import { useDocumentProvider } from '../hooks/useDocumentProvider';
+import { useDocumentRelationshipProvider } from '../hooks/useDocumentRelationshipProvider';
+import DocumentUploadView from './DocumentUploadView';
+
+export interface IDocumentUploadContainerProps {
+  parentId: number;
+  relationshipType: DocumentRelationshipType;
+  onUploadSuccess: () => void;
+  onCancel: () => void;
+}
+
+export const DocumentUploadContainer: React.FunctionComponent<IDocumentUploadContainerProps> = props => {
+  const isMounted = useIsMounted();
+  const { getDocumentTypes } = useApiDocuments();
+  const { retrieveDocumentMetadata, retrieveDocumentMetadataLoading } = useDocumentProvider();
+
+  const { uploadDocument, uploadDocumentLoading } = useDocumentRelationshipProvider();
+
+  const [documentTypes, setDocumentTypes] = useState<Api_DocumentType[]>([]);
+
+  useEffect(() => {
+    const fetch = async () => {
+      const axiosResponse = await getDocumentTypes();
+      if (axiosResponse && isMounted()) {
+        setDocumentTypes(axiosResponse.data);
+      }
+    };
+
+    fetch();
+  }, [getDocumentTypes, isMounted, retrieveDocumentMetadata]);
```

Comment:
question re: useEffect dependency array - do we need to check for "retrieveDocumentMetadata" here? Doesn't seem to be in use inside the effect

Reply:
good catch. Is not needed so i've removed it